bench/blas_benchmark.hpp

```diff
@@ -0,0 +1,127 @@
+/* Copyright (c) 2015 The Khronos Group Inc.
+
+   Permission is hereby granted, free of charge, to any person obtaining a
+   copy of this software and/or associated documentation files (the
+   "Materials"), to deal in the Materials without restriction, including
+   without limitation the rights to use, copy, modify, merge, publish,
+   distribute, sublicense, and/or sell copies of the Materials, and to
+   permit persons to whom the Materials are furnished to do so, subject to
+   the following conditions:
+
+   The above copyright notice and this permission notice shall be included
+   in all copies or substantial portions of the Materials.
+
+   MODIFICATIONS TO THIS FILE MAY MEAN IT NO LONGER ACCURATELY REFLECTS
+   KHRONOS STANDARDS. THE UNMODIFIED, NORMATIVE VERSIONS OF KHRONOS
+   SPECIFICATIONS AND HEADER INFORMATION ARE LOCATED AT
+    https://www.khronos.org/registry/
+
+  THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+  MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+  IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+  CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+  TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+  MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+*/
+
+// https://github.com/KhronosGroup/SyclParallelSTL/blob/master/benchmarks/benchmark.h
+
+#ifndef BLAS_BENCHMARK_HPP
+#define BLAS_BENCHMARK_HPP
+
+#include <chrono>
+#include <iomanip>
+#include <iostream>
+#include <string>
+#include <unistd.h>
+#include <utility>
+
+template <typename ScalarT>
+ScalarT *new_data(size_t size, bool initialized = true) {
+  ScalarT *v = new ScalarT[size];
+  if (initialized) {
+    for (size_t i = 0; i < size; ++i) {
+      v[i] = 1e-3 * ((rand() % 2000) - 1000);
+    }
+  }
+  return v;
+}
+#define release_data(ptr) delete[] (ptr);
```

Comment:
Yes, although it would do exactly the same, and there is no way I know to specify that we need a pointer that can be destroyed with `delete []`, not `delete`.